tailer/fswatcher/linereader.go

```diff
@@ -41,19 +44,32 @@ func (r *lineReader) ReadLine(file io.Reader) (string, bool, error) {
 		err error
 		buf = make([]byte, 512)
 		n   = 0
+		reg = regexp.MustCompile(r.lineDelimiter)
 	)
 	for {
-		newlinePos := bytes.IndexByte(r.remainingBytesFromLastRead, '\n')
-		if newlinePos >= 0 {
+		newlinesLoc := reg.FindAllIndex(r.remainingBytesFromLastRead, 2)
```

Comment:
In almost all cases the line delimiter will be `\n` or `\r\n`, and in these cases it's overkill to evaluate regular expressions all the time. Please keep the original `bytes.IndexByte()` for performance reasons if the line delimiter is `\n`, and use the regular expression only if a different value for line delimiter is configured.